File: 2937-make-three-strings-equal/2937-make-three-strings-equal.java
class Solution {
    public int findMinimumOperations(String s1, String s2, String s3) {
        int i = 0;
        
        int minLength = Math.min(s1.length(), Math.min(s2.length(), s3.length())); 
        for(;i < minLength; i++){
            if(s1.charAt(i) == s2.charAt(i) && s2.charAt(i) == s3.charAt(i)) continue;
            else break;
        }
        return i==0 ? -1 : (s1.length() + s2.length() + s3.length() - (i*3));
    }
}

File: 2937-make-three-strings-equal/2937-make-three-strings-equal.py
class Solution:
    def findMinimumOperations(self, s1: str, s2: str, s3: str) -> int:
        max_equal = 0
        for c1, c2, c3 in zip(s1, s2, s3):
            if not (c1 == c2 == c3):
                break
            max_equal += 1
        return -1 if max_equal == 0 else len(s1) + len(s2) + len(s3) - 3 * max_equal